File: solutions/zig/space-age/1/space_age.zig
pub const Planet = enum {
    mercury,
    venus,
    earth,
    mars,
    jupiter,
    saturn,
    uranus,
    neptune,

    pub fn age(self: Planet, seconds: usize) f64 {
        const earth_year_seconds: f64 = 31_557_600.0;
        return @as(f64, @floatFromInt(seconds)) / (earth_year_seconds * self.orbitalPeriod());
    }

    fn orbitalPeriod(self: Planet) f64 {
        return switch (self) {
            Planet.mercury => 0.2408467,
            Planet.venus => 0.61519726,
            Planet.earth => 1.0,
            Planet.mars => 1.8808158,
            Planet.jupiter => 11.862615,
            Planet.saturn => 29.447498,
            Planet.uranus => 84.016846,
            Planet.neptune => 164.79132,
        };
    }
};

File: solutions/c/space-age/1/space_age.c
#include "space_age.h"
#define EARTH_SECONDS 31557600

float orbital_periods[] = { 0.2408467f, 0.61519726f, 1.0f, 1.8808158f, 11.862615f, 29.447498f, 84.016846f, 164.79132f };
float age(planet_t planet, int64_t seconds) {
  if ((int) planet < 0) return -1.0f;
  return seconds / (orbital_periods[planet] * EARTH_SECONDS);
}
